pkg/cloudproduct/cloudproduct.go

```diff
@@ -48,7 +48,7 @@ type ControllerHooksInterface interface {
 	SyncPodPortsToGameServer(*agonesv1.GameServer, *corev1.Pod) error
 
 	// NewPortAllocator creates a PortAllocator. See gameservers.NewPortAllocator for parameters.
-	NewPortAllocator(int32, int32, informers.SharedInformerFactory, externalversions.SharedInformerFactory) portallocator.Interface
+	NewPortAllocator(int32, int32, map[string]portallocator.PortRange, informers.SharedInformerFactory, externalversions.SharedInformerFactory) portallocator.Interface
```

Comment:
@zmerlynn can you take a peek here - since this is your area of expertise.

Reply:
The `gkeAutopilot` implementation of `APIHooks` has a `ValidateGameServerSpec` hook already: https://github.com/googleforgames/agones/blob/c1320eedf6ea21cb8e5e6a5e12005881ba725a64/pkg/cloudproduct/gke/gke.go#L134-L146 - you could reject `range` here the same way we reject everything but `Dynamic` today.

That said, we really need to reject this at _install_ time so people don't trip into this blindly. We don't have a way to do that in helm yet - do you have any ideas?

Reply:
I seem to remember some form of helm helper check that could throw an error. I would need to look for it again, but I guess that could be what we are looking for.

Reply:
The problem is the runtime check is tricky - as an example, we already do this here for the controller: https://github.com/googleforgames/agones/blob/main/pkg/cloudproduct/gke/gke.go#L78-L86 .. but doing that in helm is basically impossible (and doesn't make sense when you're just using helm as a renderer). :/ I'll keep thinking on this.

Reply:
We came to a reasonable resolution here of enforcing it at runtime where we have good contact with the user.